src/github.com/matrix-org/dendrite/clientapi/routing/profile.go

```diff
@@ -60,19 +57,40 @@ func GetProfile(
 	}
 }
 
-// GetAvatarURL implements GET /profile/{userID}/avatar_url
-func GetAvatarURL(
+func getProfileByUserID(
 	req *http.Request, accountDB *accounts.Database, userID string,
-) util.JSONResponse {
+) (*authtypes.Profile, *util.JSONResponse) {
 	localpart, _, err := gomatrixserverlib.SplitID('@', userID)
 	if err != nil {
-		return httputil.LogThenError(req, err)
+		resErr := httputil.LogThenError(req, err)
+		return nil, &resErr
 	}
 
 	profile, err := accountDB.GetProfileByLocalpart(req.Context(), localpart)
 	if err != nil {
-		return httputil.LogThenError(req, err)
+		resErr := httputil.LogThenError(req, err)
+		return nil, &resErr
 	}
+
+	if profile == nil {
+		return nil, &util.JSONResponse{
+			Code: http.StatusNotFound,
+			JSON: jsonerror.NotFound("no profile information for this user or this user does not exist"),
+		}
+	}
+
+	return profile, nil
+}
+
+// GetAvatarURL implements GET /profile/{userID}/avatar_url
+func GetAvatarURL(
```

Comment:
Should this method (and others) not also check for a `nil` profile?

Reply:
getProfileByUserID does that, part of the reason why I made this common.

Reply:
Ah I see. Misread the diff, thought you were still doing the `nil` check in `GetAvatarURL`.